CONTRIBUTING.md

```diff
@@ -1,197 +1,221 @@
 # or include the cudf and/or dask_cudf Python libraries:
 
 ./build.sh  # libcudf, cudf and dask_cudf
 ./build.sh libcudf  # libcudf only
-./build.sh libcudf cudf  # libcudf and cudf only             
+./build.sh libcudf cudf  # libcudf and cudf only
 ```
-- Other libraries like `cudf-kafka` and `custreamz` can be installed with this script. For the complete list of libraries as well as details about the script usage, run the `help` command:
+
+- Other libraries like `cudf-kafka` and `custreamz` can be installed with this script. For the
+  complete list of libraries as well as details about the script usage, run the `help` command:
+
 ```bash
-./build.sh --help            
+./build.sh --help
 ```
 
 ### Build, install and test cuDF libraries for contributors
 
-The general workflow is provided below. Please, also see the last section about [code formatting](###code-formatting).
+The general workflow is provided below. Please also see the last section about
+[code formatting](#code-formatting).
 
 #### `libcudf` (C++)
 
-If you're only interested in building the library (and not the unit tests):
- 
+- If you're only interested in building the library (and not the unit tests):
+
 ```bash
 cd $CUDF_HOME
 ./build.sh libcudf
 ```
-If, in addition, you want to build tests:
+
+- If, in addition, you want to build tests:
 
 ```bash
 ./build.sh libcudf tests
 ```
-To run the tests:
+
+- To run the tests:
 
 ```bash
-make test                                      
+make test
 ```
 
 #### `cudf` (Python)
 
 - First, build the `libcudf` C++ library following the steps above
 
-- To build and install in edit/develop `cudf` python package:
+- To build and install in edit/develop `cudf` Python package:
 ```bash
 cd $CUDF_HOME/python/cudf
 python setup.py build_ext --inplace
 python setup.py develop
 ```
 
-- To run `cudf` tests :
+- To run `cudf` tests:
 ```bash
 cd $CUDF_HOME/python
-py.test -v cudf/cudf/tests
+pytest -v cudf/cudf/tests
 ```
 
 #### `dask-cudf` (Python)
 
 - First, build the `libcudf` C++ and `cudf` Python libraries following the steps above
 
-- To install in edit/develop mode the `dask-cudf` python package:
+- To install the `dask-cudf` Python package in editable/develop mode:
 ```bash
 cd $CUDF_HOME/python/dask_cudf
 python setup.py build_ext --inplace
 python setup.py develop
 ```
 
-- To run `dask_cudf` tests :
+- To run `dask_cudf` tests:
 ```bash
 cd $CUDF_HOME/python
-py.test -v dask_cudf
+pytest -v dask_cudf
 ```
 
 #### `libcudf_kafka` (C++)
 
-If you're only interested in building the library (and not the unit tests):
- 
+- If you're only interested in building the library (and not the unit tests):
+
 ```bash
 cd $CUDF_HOME
 ./build.sh libcudf_kafka
 ```
-If, in addition, you want to build tests:
+
+- If, in addition, you want to build tests:
 
 ```bash
 ./build.sh libcudf_kafka tests
 ```
-To run the tests:
+
+- To run the tests:
 
 ```bash
-make test                                      
+make test
```

Comment:
It would be useful if there was an example of how to run just one single test, for c++ and python. I rarely work in the c++ layer and always have to poke around slack to figure out how to run just one c++ test when needed.

Reply:
Yes! I started this PR specifically to address concerns like:
- running a single test
- running a single benchmark
- running all benchmarks
- building docs

However, I think all of these topics would fit better in the C++ [Developer Guide](https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/docs/DEVELOPER_GUIDE.md), [Benchmarking Guide](https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/docs/BENCHMARKING.md), [Documentation Guide](https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/docs/DOCUMENTATION.md), and [Testing Guide](https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/docs/TESTING.md). Would it solve the problem if we cross-linked one or more of those documents here and did corresponding updates to those docs?

Reply:
See also #9251 which has similar ideas for cross-linking.

Reply:
As long as it makes it in somewhere I'm happy.

Reply:
I am going to leave this conversation open since it contains action items for future PRs. However, it would be nice to get a PR approval if you think the rest is good to go.